builtin/providers/aws/resource_aws_autoscaling_lifecycle_hook.go

```diff
@@ -55,14 +59,20 @@ func resourceAwsAutoscalingLifecycleHook() *schema.Resource {
 }
 
 func resourceAwsAutoscalingLifecycleHookPut(d *schema.ResourceData, meta interface{}) error {
-	autoscalingconn := meta.(*AWSClient).autoscalingconn
-
 	params := getAwsAutoscalingPutLifecycleHookInput(d)
 
 	log.Printf("[DEBUG] AutoScaling PutLifecyleHook: %#v", params)
-	_, err := autoscalingconn.PutLifecycleHook(&params)
+	stateConf := &resource.StateChangeConf{
+		Pending:    []string{"retrying"},
+		Target:     "success",
+		Refresh:    resourceAwsAutoscalingLifecycleHookRefreshFunc(meta, params),
+		Timeout:    1 * time.Minute,
+		MinTimeout: 3 * time.Second,
+	}
```

Comment:
I think in simple cases like these we might be able to get simpler code up at this layer by leaning on the `resource.Retry` helper, which takes care of the boilerplate: https://github.com/hashicorp/terraform/blob/master/helper/resource/wait.go#L12

What do you think, @catsby?